internal/idler/user_idler_test.go

```diff
@@ -84,23 +85,117 @@ func Test_timeout_occurs_regardless_of_other_events(t *testing.T) {
 	wg.Wait()
 
 	logMessages := extractLogMessages(hook.Entries)
-	idleAftercount := 0
+	idleAfterCount := 0
 	userDataCount := 0
 	for _, message := range logMessages {
-		if message == "IdleAfter timeout." {
-			idleAftercount++
+		if message == "Time based idle check." {
+			idleAfterCount++
 		}
 		if message == "Received user data." {
 			userDataCount++
 		}
 	}
 
-	assert.Equal(t, 3, idleAftercount, "The timeout should have occurred 3 times.")
+	assert.Equal(t, 3, idleAfterCount, "The timeout should have occurred 3 times.")
 	assert.Equal(t, 2, userDataCount, "User data should have been received twice")
 
 	assert.Contains(t, logMessages, "Shutting down user idler.", "NNo proper shutdown recorded.")
 }
 
+func Test_number_of_idle_calls_are_capped(t *testing.T) {
```

Comment:
We really get into a state now were one can actually write unit tests for isolated parts of the Idler :-)